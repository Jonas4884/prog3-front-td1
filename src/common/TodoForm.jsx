import { useState } from 'react';
import { TodoStatus } from './utils/status';



const TodoForm = (props) => {
  const [Todo, setTodoValue] = useState('');
  const { add } = props;

  const handleSubmit = (event) => {
    event.preventDefault();
    if (Todo.length > 0) {
      add({ id: new Date().getTime(), status: TodoStatus.TODO, value: Todo });
      setTodoValue('');
    }
  };

  const handleChange = (event) => {
    setTodoValue(event.target.value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input  className='todo__input' placeholder='Add new task' name='name' id='name' value={Todo} type='text' onChange={handleChange} />
        <label htmlFor="name" className='form__label'>Press enter to add Task</label>
    </form>
  );
};

export default TodoForm;
