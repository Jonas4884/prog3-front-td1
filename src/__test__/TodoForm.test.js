import { fireEvent, render, screen } from "@testing-library/react";
import userEvent, { user } from "@testing-library/user-event";
import TodoForm from "../common/TodoForm";

describe("should return the todo add task", () => {
  test("should the input placeholder", () => {
    render(<TodoForm />);
    const thePlaceholder = screen.getByPlaceholderText("Add new task");
    expect(thePlaceholder).toBeInTheDocument();
  });

  test("should return a label value", () => {
    render(<TodoForm />);
    const toHaveaLabel = screen.getByLabelText("Press enter to add Task");
    expect(toHaveaLabel).toBeInTheDocument();
  });

  test("should return a value after focusing on input", () => {
    const addTodo = "jadd okoko";

    render(<TodoForm add={[addTodo]} />);
    const thePlaceholder = screen.getByPlaceholderText("Add new task");
    expect(thePlaceholder).toBeInTheDocument();
  });
 
  test("should can focus on the blank js test", () => {
    render(<TodoForm />);
    const input = document.querySelector("input")
    input.focus();
    expect(document.activeElement).toBe(input);
  });

  test("should can input on the blank", () => {
    render(<TodoForm />);
    const thePlaceholder = screen.getByPlaceholderText("Add new task");
    userEvent.type(thePlaceholder, "addnew todo");
    expect(thePlaceholder).toHaveValue();
  });

  test("should disapear the input on enter typing", () => {
    const addTodo = "jadd okoko";
    render(<TodoForm add={[]} />);
    const input = screen.getByRole("textbox");
    fireEvent.change(input, { target: { value: addTodo } });
    fireEvent.keyDown(input, { key: "Enter", code: 13 });
    expect(input.innerHTML).toBe("");
  });
});
