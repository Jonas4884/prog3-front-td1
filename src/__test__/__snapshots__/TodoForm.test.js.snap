// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return the todo add task should map a snapshot 1`] = `
<form
  onSubmit={[Function]}
  test-id="the__item"
>
  <input
    className="todo__input"
    data-testid="element-to-focus"
    id="name"
    name="name"
    onChange={[Function]}
    placeholder="Add new task"
    type="text"
    value=""
  />
  <label
    className="form__label"
    htmlFor="name"
  >
    Press enter to add Task
  </label>
</form>
`;
